kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/ranges/Range.kt

```diff
@@ -0,0 +1,70 @@
+package io.kotest.matchers.ranges
+
+data class Range<T: Comparable<T>>(
+    val start: RangeEdge<T>,
+    val end: RangeEdge<T>
+) {
+   init {
+      require(start.value <= end.value) {
+         "${start.value} cannot be after ${end.value}"
+      }
+   }
+
+   override fun toString(): String {
+      return "${if(start.edgeType == RangeEdgeType.INCLUSIVE) "[" else "("}${start.value}, ${end.value}${if(end.edgeType == RangeEdgeType.INCLUSIVE) "]" else ")"}"
```

Comment:
Complicated, but I like being mathematically correct! Kudos for using the symbols correctly.

I've also seen "[1, 3[" being used instead of "[1,3)". I think either way is fine